### ISSUE TITLE
Missing CREATE API INTEGRATION in Snowflake

### ISSUE THREAD

**Issue by moreaupascal56** [sqlfluff/sqlfluff#6814]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The CREATE API INTEGRATION clause is missing in snowflake. 

https://docs.snowflake.com/en/sql-reference/sql/create-api-integration

Example:

```
➜  ~ sqlfluff lint --dialect snowflake tmp.sql
== [tmp.sql] FAIL
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE OR
                       | REPLACE API INTEGRATION demons...'
L:   6 | P:   1 | LT12 | Files must end with a single trailing newline.
                       | [layout.end_of_file]
WARNING: Parsing errors found and dialect is set to 'snowflake'. Have you configured your dialect correctly?
All Finished 📜 🎉!
➜  ~ sqlfluff --version
sqlfluff, version 3.3.1
➜  ~ cat tmp.sql
CREATE OR REPLACE API INTEGRATION demonstration_external_api_integration_01
  API_PROVIDER = aws_api_gateway
  API_AWS_ROLE_ARN = 'arn:aws:iam::123456789012:role/my_cloud_account_role'
  API_ALLOWED_PREFIXES = ('https://xyz.execute-api.us-west-2.amazonaws.com/production')
  ENABLED = TRUE;

```

### Expected Behaviour

Should be able to lint the file 

### Observed Behaviour

see above

### How to reproduce

try to use sqlfluff with this example 

```sql

CREATE OR REPLACE API INTEGRATION demonstration_external_api_integration_01
  API_PROVIDER = aws_api_gateway
  API_AWS_ROLE_ARN = 'arn:aws:iam::123456789012:role/my_cloud_account_role'
  API_ALLOWED_PREFIXES = ('https://xyz.execute-api.us-west-2.amazonaws.com/production')
  ENABLED = TRUE;

```

### Dialect

snowflake

### Version

version 3.3.1

### Configuration

no config

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
